test/FunctionalTests/Microsoft.OData.Core.Tests/JsonLight/ODataJsonLightWriterTests.cs

```diff
@@ -1394,6 +1394,87 @@ public async Task WriteDeletedResourceAsync_WorksForDeletedResourceAtTopLevel()
                 result);
         }
 
+        [Fact]
+        public async Task WriteTransientComplexResourceAsync_ODataIdAnnotationIsNotWritten()
+        {
+            var orderResource = new ODataResource
+            {
+                TypeName = "NS.Order",
+                Properties = new List<ODataProperty>(),
+                IsTransient = true,
+                SerializationInfo = CreateOrderResourceSerializationInfo()
+            };
+
+            var addressNestedResourceInfo = CreateAddressNestedResourceInfo();
+            var addressResource = new ODataResource
+            {
+                TypeName = "NS.Address",
+                Properties = new List<ODataProperty>
+                {
+                    new ODataProperty { Name = "City", Value = "Redmond" }
+                },
+                IsTransient = true,

```

Comment:
How should the output be if `IsTransient = false`?

Reply:
@KenitoInc If `IsTransient` has a value of `false` it shouldn't matter in the case of a complex type. But in the case of an entity, an exception would be thrown if the key property is missing.

What was happening before however was that `IsTransient` with a value of `true` was causing `"odata.id":null` property to be written in the payload even where the OData resource was a complex type.